src/WorkerPool.js

```diff
@@ -186,7 +187,19 @@ class PoolWorker {
     const err = new Error(obj.message);
     err.message = obj.message;
     if (obj.stack) {
-      err.stack = `${obj.stack}\n\tfrom thread-loader (worker ${this.id})\n${err.stack}`;
+      err.stack = `${obj.stack}\n\tThread Loader (Worker ${this.id})`;
+      if (this.options.stack) {
+        // Limit the err.stack manually in case stackTraces are to verbose
+        if (typeof this.options.stack === 'number') {
+          Error.stackTraceLimit = (this.options.stack);
```

Comment:
Is this setting a global? It seems very non-intuitive to have a package set global configuration like this.

Reply:
good one. But I dont think this will have a global impact. But @michael-ciniawsky would be correct one to answer

Reply:
You're setting it on a global so I don't see how it could be local.
From the docs (https://nodejs.org/api/errors.html#errors_error_stacktracelimit):

>Changes will affect any stack trace captured after the value has been changed.

Reply:
But it changes the value in a worker thread.  ( well I may be totally wrong) I will check this

Reply:
This is the code that *rethrows* the error from the worker so it runs in the main thread.
I suggested another solution below.